openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -67,15 +67,16 @@ def process_response(self, request, response):
                     # If we can't find the user preferences, then don't modify the cookie
                     pass
 
-            # If set, set the user_pref in the LANGUAGE_COOKIE
-            if user_pref:
+            # If set and user is not a masquerading user, set the user_pref in the LANGUAGE_COOKIE
+            is_masquerading_user = request.session.get(MASQUERADE_SETTINGS_KEY, {})
+            if user_pref and not is_masquerading_user:
```

Comment:
What about doing something like:
```python
if user_pref:
    if not is_masquerading_user:
```
this way we will be able to save next `if` on line # 79